include/boost/json/detail/impl/number.ipp

```diff
@@ -396,6 +447,10 @@ loop:
             }
             if(*p == 'e' || *p == 'E')
             {
+                // if we have not encountered a decimal point then
+                // we must assume it's at the [eE]
+                // because 100e0 is equivalent to 100.0e0
+                if (pos_ < 0) pos_ = dig_;
```

Comment:
The condition and statements need to be on separate lines for coverage and breakpoints